litmus-portal/frontend/src/components/CommunityTimeSeriesTrendPlot/index.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { Component } from 'react';
+import Plotly from 'plotly.js';
+import createPlotlyComponent from 'react-plotly.js/factory';
+
+const Plot = createPlotlyComponent(Plotly);
+
+type GraphState = {
+  x: any[];
+  y: any[];
+  y2: any[];
+};
+
+interface CommunityAnalyticsPlotProps {
+  OperatorData?: any;
+  ExperimentData?: any;
+}
+
+class CommunityAnalyticsPlotNormal extends Component<
+  CommunityAnalyticsPlotProps,
+  GraphState
+> {
+  constructor(props: Readonly<CommunityAnalyticsPlotProps>) {
+    super(props);
+    this.state = {
+      x: [],
+      y: [],
+      y2: [],
+    };
+    this.processData = this.processData.bind(this);
+  }
+
+  componentDidMount() {
+    this.processData();
+  }
+
+  processData() {
+    const rawData: any = [];
+    const { OperatorData, ExperimentData } = this.props;
+    OperatorData.forEach((element: any, index: any) => {
+      rawData.push({
+        date: element.date,
+        operatorInstalls: element.count,
+        experimentRuns: ExperimentData[index].count,
+      });
+    });
+
+    const x: Date[] = [];
+    const y: string[] = [];
+    const y2: string[] = [];
+    rawData.forEach(function dateSplit(datum: any) {
+      const splits = datum.date.split('-');
+      x.push(
+        new Date(splits[0] as any, (splits[1] as any) - 1, splits[2] as any)
+      );
+      y.push(datum.operatorInstalls);
+      y2.push(datum.experimentRuns);
+    });
+    this.setState({ x });
+    this.setState({ y });
+    this.setState({ y2 });
+  }
+
+  render() {
+    const selectorOptions = {
+      buttons: [
+        {
+          step: 'day',
+          stepmode: 'backward',
+          count: 7,
+          label: '1w',
+        },
+        {
+          step: 'day',
+          stepmode: 'backward',
+          count: 14,
+          label: '2w',
+        },
+        {
+          step: 'month',
+          stepmode: 'backward',
+          count: 1,
+          label: '1m',
+        },
+        {
+          step: 'month',
+          stepmode: 'backward',
+          count: 3,
+          label: '3m',
+        },
+        {
+          step: 'month',
+          stepmode: 'backward',
+          count: 6,
+          label: '6m',
+        },
+        {
+          step: 'year',
+          stepmode: 'backward',
+          count: 1,
+          label: '1y',
+        },
+        {
+          step: 'all',
+          label: 'Default',
+        },
+      ],
+    };
+
+    const { x } = this.state;
```

Comment:
destructure all variables at once...

```suggestion
    const {x, y, y2} = this.state;
```